### PR TITLE
Stop warning on .names() access in max_pool2d and max_pool2d_backward

### LINKED ISSUE #60053: Annoying warning with nn.MaxPool2d
Hi, thank you very much for pytorch 1.9!
I'm trying to update SpeechBrain (https://github.com/speechbrain/speechbrain) to support pytorch 1.9.
Everything seems to work, but I noticed an annoying warning when using  nn.MaxPool2d:

```
import torch
import torch.nn as nn

m = nn.MaxPool2d(3, stride=2)
m = nn.MaxPool2d((3, 2), stride=(2, 1))
input = torch.randn(20, 16, 50, 32)
output = m(input)
```

```
UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /opt/conda/conda-bld/pytorch_1623448238472/work/c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
```

Note that it only appears when using nn.MaxPool2d (nn.MaxPool1d and  nn.AvgPool1d have no warning).
Any idea?

cc @zou3519 @albanD @mruberry @jbschlosser

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60059 Stop warning on .names() access in max_pool2d and max_pool2d_backward**

Fixes #60053.

The problem is that `.names()` always triggers the named tensor warning.
To not trigger it, one has to guard it with has_names:
`x.has_names() ? x.names() : DimnameList{}`

This is not the first time this has happened; we should probably
make it so that .names() doesn't raise a warning unless it is actually
populated with names. That's a little tricky to implement so I'm leaving
it for the future.

Test Plan:
- New test, also run `python test/test_nn.py -v -k "max_pool"` and
confirm there are no warnings.

Differential Revision: [D29152737](https://our.internmc.facebook.com/intern/diff/D29152737)

### CODE REVIEW

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29152737).

File: test/test_namedtensor.py [hunk 1]
```diff
@@ -296,6 +296,14 @@ def check_tuple_return(op, inputs, expected_names):
             check_tuple_return(F.max_pool2d_with_indices, [named_tensor_2d, [2, 2]], named_tensor_2d.names)
             check_tuple_return(F.max_pool3d_with_indices, [named_tensor_3d, [2, 2, 2]], named_tensor_3d.names)
 
+    def test_max_pooling_without_names_does_not_warn(self):
```
Comment (gchanan) [inline test/test_namedtensor.py:299]: test cuda

**Comment by gchanan** [comment]
do we have a check that the names are checked correctly?  I don't see where that would be happening.

**Comment by zou3519** [comment]
> do we have a check that the names are checked correctly? I don't see where that would be happening.

There's no name checking in max_pool IIRC: because max_pool accepts a single tensor, the behavior is that the result tensor should have the same names as the input tensor. (We didn't decide to e.g. enforce that the input to max_pool2d has names N C H W)

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29152737).

**Review by gchanan — APPROVED** [review]

**Pull request closed without merging**